### PR TITLE
test: Build with Eask

### PR DESCRIPTION
**Pull request by jcs090218**
Add basic tests using Eask. 

Current tests:

- Test packaging (into `tar`)
- Test install
- Test byte-compile

See the log in https://github.com/jcs-PR/nim-mode/actions/runs/6344019152?pr=1.

There is an error in the Emacs windows snapshot, and we should ignore that for now. I've added the `experimental` flag so the badge will remain green [CI](https://github.com/jcs-PR/nim-mode/actions/workflows/test.yml).

### CODE REVIEW

File: nim-mode.el [hunk 1]
```diff
@@ -170,7 +170,7 @@
   (when (and (derived-mode-p 'org-mode)
              (fboundp 'org-in-src-block-p) (org-in-src-block-p))
     (modify-syntax-entry ?# ". 124b" nim-mode-syntax-table)
-    (modify-syntax-entry ?[ ". 23" nim-mode-syntax-table))
+    (modify-syntax-entry ?\[ ". 23" nim-mode-syntax-table))
```
Comment (Vindaar) [inline nim-mode.el:173]: Ohh, I see! Another instance of me being confused as to why emacs is happy with a non escaped `[` there. I've got a lot to learn. :laughing:

File: Makefile [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
 EMACS ?= emacs
 
-cask_dir_exists := $(shell find ./ -type d -name ".cask")
-
 all: test
 
-test: clean-elc
-	$(if $(cask_dir_exists), $(skip), ${MAKE} install-dependencies)
-	${MAKE} unit
+# TODO: Add `test` when it's stable!
```
Comment (Vindaar) [inline Makefile:6]: Can you expand on what you mean by stable here? The tests in the repo fail currently?

File: nim-mode.el (hunk 1 above)
Comment (jcs090218) [reply nim-mode.el:173]: Yeah, without the escaped you will see

```
nim-mode.el:186:84: Error: Invalid read syntax: ")", 186, 84
```

See log in https://github.com/jcs-PR/nim-mode/actions/runs/6343881032/job/17232745497#step:5:119..

File: Makefile (hunk 2 above)
Comment (jcs090218) [reply Makefile:6]: Yes, the tests are failing. I don't have the knowledge regarding the tests in this repo, so I have temporarily disabled it. I recommend opening another PR to fix the tests. 🤔 

See log in https://github.com/jcs-PR/nim-mode/actions/runs/6343959389/job/17233002351.

File: nim-mode.el (hunk 1 above)
Comment (Vindaar) [reply nim-mode.el:173]: Huh. What's going on with my emacs installation that it doesn't trigger that? :rofl: Thanks!

File: Makefile (hunk 2 above)
Comment (Vindaar) [reply Makefile:6]: Ok, fair enough! Once I have more time I'll try to take a look at them.

**Comment by Vindaar** [comment]
Thanks a lot, this is really very much appreciated. :heart:

File: nim-mode.el (hunk 1 above)
Comment (jcs090218) [reply nim-mode.el:173]: Only Emacs lower than 28 is triggering this error, but I have no idea why. That's why it's good to have CI tests. ;)

**Pull request merged**